java/code/src/com/redhat/rhn/taskomatic/task/RecurringStateApplyJob.java

```diff
@@ -57,9 +58,9 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
     }
 
     private void scheduleAction(JobExecutionContext context, RecurringAction action) {
-        List<Long> minionIds = action.computeMinions().stream()
-                .map(m -> m.getId())
-                .collect(Collectors.toList());
+        List<Long> minionIds = MaintenanceManager.systemIdsMaintenanceMode(action.computeMinions().stream()
```

Comment:
The stream -> collect is not necessary. You can feed `action.computeMinions()` directly to `systemidsMaintenanceMode`.